modules/lsp/server.nix

```diff
@@ -70,5 +68,23 @@ in
           ];
         };
     };
+
+    # NOTE: we need a warnings option for `mkRenamedOptionModule` to warn about unexpected definitions
+    # This can be removed when all rename aliases are gone
+    warnings = lib.mkOption {
+      type = with types; listOf str;
+      description = "Warnings to propagate to nixvim's `warnings` option.";
+      default = [ ];
+      internal = true;
+      visible = false;
+    };
   };
+
+  imports = [
+    # TODO: rename added 2025-04-30 (during the 25.05 cycle)
+    # The previous name `config` was introduced 2025-04-28 (during the 25.05 cycle)
+    # Because the previous name `config` never made it into a stable release,
+    # we could consider dropping this alias sooner than normal.
+    (lib.mkRenamedOptionModule [ "config" ] [ "settings" ])
```

Comment:
Was gonna say it wasn't alive long enough to really need a rename.. but i supose to we can be nice :P

Reply:
Maybe, but we've already posted an announcement in matrix and had feedback from users who are already using the new module.

On _that_ basis I figured a short-term warning was worth the small amount of effort.